### PR TITLE
update pip 8.0.0 and its complication on Travis

### PR DESCRIPTION
**Pull request by kitchoi**
pip is upgraded to 8.0.0 and the link to get-pip.py needs to be updated.

The site packages ``argparse`` and ``pyyaml`` on Travis are installed by distutils and the new pip refuses to upgrade them.  So I uninstall and reinstall them using ``sudo pip``, which would load the site's 7.1.2 ``pip`` instead of the installed 8.0.0 one.

If you wonder why not just use pip 7.1.2 for everything at the very beginning, I tried, but it won't concede: https://travis-ci.org/simphony/simphony-framework/jobs/103801474

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -244,7 +244,7 @@ simphony-plugins: simphony-kratos simphony-numerrin simphony-mayavi simphony-ope
 	@echo
 	@echo "Simphony plugins installed"
 
-simphony-framework:
+simphony-framework: simphony simphony-plugins
```
Comment (itziakos) [inline Makefile]: why is this change necessary?

File: .travis.yml [hunk 2]
```diff
@@ -19,7 +19,10 @@ before_install:
   - sudo make base
   - sudo make apt-openfoam-deps apt-simphony-deps apt-lammps-deps apt-mayavi-deps apt-aviz-deps
   - make fix-pip
-  - pip install --allow-external PIL --allow-unverified PIL PIL
```
Comment (itziakos) [inline .travis.yml]: This line should have never been here in the first place

File: .travis.yml [hunk 3]
```diff
@@ -19,7 +19,10 @@ before_install:
...
+  - sudo pip uninstall -y argparse
```
Comment (itziakos) [inline .travis.yml]: pip install --upgrade argparse
```

inside the simphony-common target?

Comment (itziakos) [inline .travis.yml]: In anycase if these commands are necessary they should be part of the fix-pip or the related target that requires them

Comment (kitchoi) [inline .travis.yml]: Yes I have tried ``pip install --upgrade argparse`` inside the simphony-common target.  In that case it would fail as the ``pip`` (in the virtual environment) complains about not knowing how to uninstalled an ``argparse`` package installed by ``distutil``.  I believe this is precisely why ``make simphony`` fails in the first place as it tries to update ``argparse``

I am not sure if ``sudo pip`` reinstall ``argparse`` is necessary generally.  I suspect this is a problem specific to packages come with Ubuntu 12.04 LTS.  Need to investigate more to find out where the dist-package's ``argparse`` come from.

File: Makefile (hunk 1 above)
Comment (kitchoi) [inline Makefile]: just thought it is a bit misleading to print "simphony framework installed" when it actually does nothing

Comment (kitchoi) [inline Makefile]: Should I separate it to a different PR?

File: .travis.yml (hunk 3 above)
Comment (itziakos) [inline .travis.yml]: For the tracebacks it looks that the new numpy has a problem with argprase. You might want to do `sudo pip install --upgrade argparse` in `fix-pip` or  `sudo apt-get remove python-argparse` in `apt-symphony`.

Comment (itziakos) [inline .travis.yml]: correction in this case `sudo pip -install --upgrade argparse` in `fix-pip` should be the right thing to do. Maybe rename `fix-pip` -> `fix-system-python`

**Comment by kitchoi** [comment]
Aha!  Look at this:
> PIP Release Note
> 8.0.1 (2016-01-21)
> Upgrading a distutils installed item that is installed outside of a virtual environment, while inside of a virtual environment will no longer warn or error.

**Comment by kitchoi** [comment]
@itziakos PIP rolled back on their decision to remove the ability to uninstall distutils installed items until a future date.  Now the PR is trivial.

**Comment by itziakos** [comment]
@kitchoi, since it looks that test-framework tests everithing and there is no time out we can probably remove the second build setup. Other than that :+1:

**Comment by kitchoi** [comment]
Yeah.  Merging.
